src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -27,7 +27,7 @@ public class SnippetsProvider : ISnippetsProvider
         // Used to cache resource dependencies. Maps resource type to it's dependencies
         private ConcurrentDictionary<string, string> resourceTypeToDependentsMap = new ConcurrentDictionary<string, string>();
         // Used to cache resource body snippets
-        private ConcurrentDictionary<string, IEnumerable<Snippet>> resourceBodySnippetsCache = new ConcurrentDictionary<string, IEnumerable<Snippet>>();
+        private ConcurrentDictionary<(string, bool), IEnumerable<Snippet>> resourceBodySnippetsCache = new ConcurrentDictionary<(string, bool), IEnumerable<Snippet>>();
```

Comment:
Can you make those properties readonly, use named tuple so there will be clear understanding what each type is for and perhaps use just `new` operator?

Reply:
Good pt! The language server project targets.net 5 so you can use c# records to make it simpler.

Reply:
Makes sense. Updated here:
https://github.com/Azure/bicep/commit/46f8ab1a0a6e2f937d3d1f167885e80edeff77e1

Thanks!

Reply:
Can you remind me if named tuples are mutable in c#? If yes, then the key stored in the dictionary could be modified after it's stored in it.

Reply:
At first I wanted to say no, but that is for classic Tuples. ValueTuples are mutable: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples